### PR TITLE
Static link token

### PR DESCRIPTION
**Pull request by connorwstein**
Supporting viewing non-B/M link token. Side quest backfilling some tests and improving input validation.

### CODE REVIEW

File: deployment/common/changeset/state.go [hunk 1]
```diff
@@ -66,29 +70,51 @@ func (state MCMSWithTimelockState) GenerateMCMSWithTimelockView() (v1_0.MCMSWith
...
+	canceller := deployment.NewTypeAndVersion(types.CancellerManyChainMultisig, deployment.Version1_0_0)
+	bypasser := deployment.NewTypeAndVersion(types.BypasserManyChainMultisig, deployment.Version1_0_0)
+
+	// Ensure we either have the bundle or not.
+	_, err := deployment.AddressesContainBundle(addresses,
+		map[deployment.TypeAndVersion]struct{}{
+			timelock: {}, proposer: {}, canceller: {}, bypasser: {},
+		})
+	if err != nil {
+		return nil, err
```
Comment (AnieeG) [inline deployment/common/changeset/state.go]: how about wrapping error with chain.String()

File: deployment/address_book.go [hunk 2]
```diff
@@ -271,3 +271,26 @@ func AddressBookContains(ab AddressBook, chain uint64, addrToFind string) (bool,
...
+// contains a single instance of all the addresses in the bundle.
+// It returns an error if there are more than one instance of a contract.
+func AddressesContainBundle(addrs map[string]TypeAndVersion, wantTypes map[TypeAndVersion]struct{}) (bool, error) {
+	counts := make(map[TypeAndVersion]int)
+	for wantType := range wantTypes {
+		for _, haveType := range addrs {
+			if wantType == haveType {
+				counts[wantType]++
+				if counts[wantType] > 1 {
+					return false, fmt.Errorf("found more than one instance of a contract")
```
Comment (AnieeG) [inline deployment/address_book.go]: can we show the type in error message? 
Wdyt about using multierror and showing all types which will have more than one instance?

File: deployment/common/changeset/state.go [hunk 3]
```diff
@@ -110,10 +136,17 @@ func (s LinkTokenState) GenerateLinkView() (v1_0.LinkTokenView, error) {
...
 }
 
-func LoadLinkTokenState(chain deployment.Chain, addresses map[string]deployment.TypeAndVersion) (*LinkTokenState, error) {
+func MaybeLoadLinkTokenState(chain deployment.Chain, addresses map[string]deployment.TypeAndVersion) (*LinkTokenState, error) {
 	state := LinkTokenState{}
+	linkToken := deployment.NewTypeAndVersion(types.LinkToken, deployment.Version1_0_0)
+	// Perhaps revisit if we have a use case for multiple.
+	_, err := deployment.AddressesContainBundle(addresses, map[deployment.TypeAndVersion]struct{}{linkToken: {}})
+	if err != nil {
+		return nil, err
```
Comment (AnieeG) [inline deployment/common/changeset/state.go]: err wrap with chain details

File: deployment/common/view/v1_0/static_link_token.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+type StaticLinkTokenView struct {
+	types.ContractMetaData
+	Decimals uint8    `json:"decimals"`
+	Supply   *big.Int `json:"supply"`
+}
+
+func GenerateStaticLinkTokenView(lt *link_token_interface.LinkToken) (StaticLinkTokenView, error) {
+	decimals, err := lt.Decimals(nil)
+	if err != nil {
+		return StaticLinkTokenView{}, fmt.Errorf("failed to get decimals: %w", err)
```
Comment (AnieeG) [inline deployment/common/view/v1_0/static_link_token.go]: link token address in err

**Review by AnieeG — COMMENTED** [review]
Generic suggestion about wrapping the error with address and chain details whenever possible

**Review by ecPablo — DISMISSED** [review]
LGTM!

File: deployment/address_book.go (hunk 2 above)
Comment (connorwstein) [reply deployment/address_book.go]: type for sure. multierr maybe bit overkill

**Review by AnieeG — APPROVED** [review]

File: deployment/common/types/types.go [hunk 5]
```diff
@@ -16,7 +16,16 @@ const (
...
+	// LinkToken is the burn/mint link token. It should be used everywhere for
+	// new deployments. Corresponds to
+	// https://github.com/smartcontractkit/chainlink/blob/develop/core/gethwrappers/shared/generated/link_token/link_token.go#L34
+	LinkToken deployment.ContractType = "LinkToken"
+	// StaticLinkToken represents the (very old) non-burn/mint link token.
+	// It is not used in new deployments, but still exists on some chains
+	// and has a distinct ABI from the new LinkToken.
+	// Corresponds to the ABI
+	// https://github.com/smartcontractkit/chainlink/blob/develop/core/gethwrappers/generated/link_token_interface/link_token_interface.go#L34
+	StaticLinkToken deployment.ContractType = "StaticLinkToken"
```
Comment (krehermann) [inline deployment/common/types/types.go:28]: maybe LegacyLinkToken?

**Review by krehermann — APPROVED** [review]

**Pull request merged**